java/ql/src/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -27,10 +27,44 @@ class AndroidReceiveIntentMethod extends Method {
   }
 }
 
-class ContextStartActivityMethod extends Method {
-  ContextStartActivityMethod() {
-    (hasName("startActivity") or hasName("startActivities")) and
-    getDeclaringType() instanceof TypeContext
+/** A constructor of `android.content.Intent`. */
+class IntentConstructor extends TaintPreservingCallable {
+  IntentConstructor() { this.(Constructor).getDeclaringType() instanceof TypeIntent }
+
+  override predicate returnsTaintFrom(int arg) {
+    arg = 0 and this.getParameterType(arg) instanceof TypeIntent
+    or
+    arg in [1, 4] and this.getParameterType(arg) instanceof TypeClass
```

Comment:
Is there a difference between using `in [1, 4]` and `= [1, 4]` here?

Reply:
No. `in` is syntactic sugar and simply means `=`. The reason for this is historical: `<expr> in [ <expr> .. <expr> ]` used to be its own syntactic construct, but with the introduction of the `[ <expr> (, <expr> )* ]` syntax the two `[ ... ]` constructs were elevated to proper expressions, which meant that `in` became synonymous to `=`.